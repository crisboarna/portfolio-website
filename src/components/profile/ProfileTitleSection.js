import React from 'react';
import ScrollAnimation from 'react-animate-on-scroll';

import '../../styles/profile/ProfileTitleSection.css';

const ProfileTitleSection = () => {
  return (
    <div className="row col-lg-12 text-center">

      <ScrollAnimation animateIn='fadeInLeft' animateOnce={true} offset={50}>
        <div className='profileTitle'>PROFILE</div>
      </ScrollAnimation>
      <ScrollAnimation animateIn='fadeInLeft' animateOnce={true}  delay={500}>
        <div><hr/></div>
      </ScrollAnimation>

      <br/><br/>

      <div className='container'>
        <div className="row col-lg-12">
          <div className="col-xs-6 col-sm-offset-2 col-sm-2">
            <ScrollAnimation animateIn='fadeIn' animateOnce={true} delay={300}>
              <div className='propertyTitle'>Design.</div>
            </ScrollAnimation>
          </div>
          <div className="col-xs-6 col-sm-2">
            <ScrollAnimation animateIn='fadeIn' animateOnce={true} delay={500}>
              <div className='propertyTitle'>Delivery.</div>
            </ScrollAnimation>
          </div>
          <div className="col-xs-6 col-sm-2">
            <ScrollAnimation animateIn='fadeIn' animateOnce={true} delay={700}>
              <div className='propertyTitle'>Back-end.</div>
            </ScrollAnimation>
          </div>
          <div className="col-xs-6 col-sm-2">
            <ScrollAnimation animateIn='fadeIn' animateOnce={true} delay={900}>
              <div className='propertyTitle'>Front-end.</div>
            </ScrollAnimation>
          </div>
        </div>
      </div>
    </div>
  );
};

export default ProfileTitleSection;