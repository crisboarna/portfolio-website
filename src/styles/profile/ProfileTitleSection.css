.profileTitle {
    font-size: 30pt;
    letter-spacing: 2px;
    color: #444649;
    text-align: center;
    font-family: "Raleway-bold";
}

.propertyTitle {
    font-family: "Raleway-bold";
    font-size: 18pt;
}

hr {
    margin: auto;
    width: 15%;
    border: 0;
    height: 0;
    box-shadow: 0 0 5px 1px black;
}