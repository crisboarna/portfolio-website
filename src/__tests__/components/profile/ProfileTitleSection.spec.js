import React from 'react';
import { shallow, mount } from 'enzyme';
import ProfileTitleSection from '../../../components/profile/ProfileTitleSection';

describe('ProfileTitleSection', () => {
  it('renders without crashing', () => {
    const component = shallow(<ProfileTitleSection />);
    expect(component.exists()).toEqual(true);
  });

  it('compare to snapshot', () => {
    const component = mount(<ProfileTitleSection/>);
    expect(component).toMatchSnapshot();
  });
});
