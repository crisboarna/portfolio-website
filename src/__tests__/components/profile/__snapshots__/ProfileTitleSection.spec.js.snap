// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileTitleSection compare to snapshot 1`] = `
<ProfileTitleSection>
  <div
    className="row col-lg-12 text-center"
  >
    <ScrollAnimation
      animateIn="fadeInLeft"
      animateOnce={true}
      delay={0}
      duration={1}
      initiallyVisible={false}
      offset={50}
    >
      <div
        className="animated"
        style={
          Object {
            "animationDuration": "1s",
            "opacity": 0,
          }
        }
      >
        <div
          className="profileTitle"
        >
          PROFILE
        </div>
      </div>
    </ScrollAnimation>
    <ScrollAnimation
      animateIn="fadeInLeft"
      animateOnce={true}
      delay={500}
      duration={1}
      initiallyVisible={false}
      offset={150}
    >
      <div
        className="animated"
        style={
          Object {
            "animationDuration": "1s",
            "opacity": 0,
          }
        }
      >
        <div>
          <hr />
        </div>
      </div>
    </ScrollAnimation>
    <br />
    <br />
    <div
      className="container"
    >
      <div
        className="row col-lg-12"
      >
        <div
          className="col-xs-6 col-sm-offset-2 col-sm-2"
        >
          <ScrollAnimation
            animateIn="fadeIn"
            animateOnce={true}
            delay={300}
            duration={1}
            initiallyVisible={false}
            offset={150}
          >
            <div
              className="animated"
              style={
                Object {
                  "animationDuration": "1s",
                  "opacity": 0,
                }
              }
            >
              <div
                className="propertyTitle"
              >
                Design.
              </div>
            </div>
          </ScrollAnimation>
        </div>
        <div
          className="col-xs-6 col-sm-2"
        >
          <ScrollAnimation
            animateIn="fadeIn"
            animateOnce={true}
            delay={500}
            duration={1}
            initiallyVisible={false}
            offset={150}
          >
            <div
              className="animated"
              style={
                Object {
                  "animationDuration": "1s",
                  "opacity": 0,
                }
              }
            >
              <div
                className="propertyTitle"
              >
                Delivery.
              </div>
            </div>
          </ScrollAnimation>
        </div>
        <div
          className="col-xs-6 col-sm-2"
        >
          <ScrollAnimation
            animateIn="fadeIn"
            animateOnce={true}
            delay={700}
            duration={1}
            initiallyVisible={false}
            offset={150}
          >
            <div
              className="animated"
              style={
                Object {
                  "animationDuration": "1s",
                  "opacity": 0,
                }
              }
            >
              <div
                className="propertyTitle"
              >
                Back-end.
              </div>
            </div>
          </ScrollAnimation>
        </div>
        <div
          className="col-xs-6 col-sm-2"
        >
          <ScrollAnimation
            animateIn="fadeIn"
            animateOnce={true}
            delay={900}
            duration={1}
            initiallyVisible={false}
            offset={150}
          >
            <div
              className="animated"
              style={
                Object {
                  "animationDuration": "1s",
                  "opacity": 0,
                }
              }
            >
              <div
                className="propertyTitle"
              >
                Front-end.
              </div>
            </div>
          </ScrollAnimation>
        </div>
      </div>
    </div>
  </div>
</ProfileTitleSection>
`;
